deepchem/models/lightning/dc_lightning_module.py

```diff
@@ -28,15 +75,40 @@ def __init__(self, dc_model):
             TorchModel to be wrapped inside the lightning module.
         """
         super().__init__()
+        self.save_hyperparameters(ignore=['dc_model'])
         self.dc_model = dc_model
-
-        self.pt_model = self.dc_model.model
-        # This will not work for ModularTorchModel as it is directly uses `loss_func` to compute loss.
-        self.loss = self.dc_model._loss_fn
+        self.pt_model = dc_model.model
+        self.loss = dc_model._loss_fn
+        self.loss_mod = dc_model.loss
+        self.optimizer = dc_model.optimizer
+        self.output_types = dc_model.output_types
+        self._prediction_outputs = dc_model._prediction_outputs
+        self._loss_outputs = dc_model._loss_outputs
+        self._variance_outputs = dc_model._variance_outputs
+        self._other_outputs = dc_model._other_outputs
+        self.uncertainty = getattr(dc_model, 'uncertainty', False)
+        self.learning_rate = dc_model.learning_rate
+        self._transformers: List[Transformer] = []
+        self.other_output_types: Optional[OneOrMany[str]] = None
 
     def configure_optimizers(self):
-        return self.dc_model.optimizer._create_pytorch_optimizer(
-            self.pt_model.parameters(),)
+        """Configure optimizers and learning rate schedulers.
+
+        Returns
+        -------
+        Union[torch.optim.Optimizer, List]
+            PyTorch optimizer or list containing optimizer and scheduler.
+        """
+        self.dc_model._built = True
+        py_optimizer = self.optimizer._create_pytorch_optimizer(
+            self.pt_model.parameters())
+
+        if isinstance(self.optimizer.learning_rate, LearningRateSchedule):
+            lr_schedule = self.optimizer.learning_rate._create_pytorch_schedule(
+                py_optimizer)
+            return [py_optimizer], [lr_schedule]
+
+        return py_optimizer
 
     def training_step(self, batch, batch_idx):
```

Comment:
please add type hints for method parameters as much as possible. Its not clear what is the type of `batch` and where is `.batch_list` coming from